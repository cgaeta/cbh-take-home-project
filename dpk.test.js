const { deterministicPartitionKey } = require("./dpk");

describe("deterministicPartitionKey", () => {
  it("Returns the literal '0' when given no input", () => {
    const trivialKey = deterministicPartitionKey();
    expect(trivialKey).toBe("0");
  });

  it("Hashes event data when event has no hash key", () => {
    const key = deterministicPartitionKey({ someData: "abc123" });
    expect(key).toMatchSnapshot();
  });

  it("Returns the event's partition key, when less that 256 characters", () => {
    const key = deterministicPartitionKey({
      partitionKey: "short_partition_key",
    });
    expect(key).toBe("short_partition_key");
  });

  it("Returns stringified partition key when partition key is an object and stringified to fewer than 256 characters", () => {
    const key = deterministicPartitionKey({
      partitionKey: { keyIsAnObject: true },
    });
    expect(key).toBe("{\"keyIsAnObject\":true}");
  });

  it("Returns a hash when the partition key is 256 characters or more", () => {
    const partitionKey = "this is a long partition key over 256 characters that should ultimately get hashed and no long be recognizable, but still be hashed determinalistically and continue to pass the test no matter how many times the test gets run with this same key, which is important so that we can test this code and make sure it doesn't break";
    const key = deterministicPartitionKey({ partitionKey });

    expect(partitionKey.length).toBeGreaterThan(256);
    expect(key).toMatchSnapshot();
  });

  it("Returns a hash when the partition key is an object that stringifies to be over 256 characters", () => {
    const partitionKey = {
      isAnObject: true,
      isUsefulData: false,
      isFewerThan256CharactersStringified: false,
      isDataThatShouldBeSeenInAProdDatabase: false,
      idealNumberOfEngineersLookingAtThisParticularTestClosely: -1,
      numberOfTimesThisTestIsExpectedToFail: 0,
      isTestedWithJestSnapshots: true,
      propertyToMakeSureThisObjectStringifiesLongEnough: "please",
    };
    const key = deterministicPartitionKey({ partitionKey });

    expect(JSON.stringify(partitionKey).length).toBeGreaterThan(256);
    expect(key).toMatchSnapshot();
  });
});
