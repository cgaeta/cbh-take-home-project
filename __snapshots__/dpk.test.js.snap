// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deterministicPartitionKey Hashes event data when event has no hash key 1`] = `"267e2edaa9a0d8a5c6a785d953253acb5c086cc0e2f8582f03c2e0c96959a72cdb97579d973e10f56ef3e2c6350609cc83417594f8253d37e19b6683e8e3481d"`;

exports[`deterministicPartitionKey Returns a hash when the partition key is 256 characters or more 1`] = `"d078a4569678c1f1680cf5b5abd40e50b1e877703a48fd05e7280fa53e1f3fdf73332993c82b44d27f6ff3f0ece047352093cf49f81f07781e209617ecff3623"`;

exports[`deterministicPartitionKey Returns a hash when the partition key is an object that stringifies to be over 256 characters 1`] = `"137a1c204d21581aab5ee4e2ce99a958f9a661728d9fefa55b1498ab9bba1dd0a66cf62a05ba8e697876aaa182eab5b6eef6ab2e6df4647c60224f14abe5421c"`;
